lib/streamlit/elements/media.py

```diff
@@ -531,6 +562,24 @@ def marshall_video(
                     f"Failed to process the provided subtitle: {label}"
                 ) from original_err
 
+    if autoplay:
+        ctx = get_script_run_ctx()
+        proto.autoplay = autoplay
+        id = compute_widget_id(
+            "video",
+            data=data,
```

Comment:
Same comment as with audio, we potentially want to use the generated URL here instead of data:

```suggestion
            url=proto.url,
```

Reply:
Fixed on 292bae7